web/src/typeahead_helper.ts

```diff
@@ -39,6 +39,17 @@ export type CombinedPillItem =
     | InputPillItem<UserGroupPill>
     | InputPillItem<StreamPill>;
 
+export type TypeaheadOptions = {
+    user: boolean;
+    user_group?: boolean;
+    stream?: boolean;
+    user_source?: () => User[];
+    exclude_bots?: boolean;
+    update_func?: () => void;
+};
+
+export type SetupTypeahead = ($input: JQuery) => void;
```

Comment:
I think it's better to keep these in `pill_typeahead.ts` where they're being used, and we can move them to a shared file in the future if they're used more widely.